src/charm.py

```diff
@@ -399,8 +404,16 @@ def _on_start(self, _):
 
     def _on_config_changed(self, _):
         """Event handler for ConfigChangedEvent."""
+        self._save_ssh_key()
         self._common_exit_hook()
 
+    def _save_ssh_key(self):
+        """Save SSH key from config to a file."""
+        ssh_key = self.config.get("git_ssh_key")
+        if ssh_key:
+            with open(self._ssh_key_file_name, "w") as output:
```

Comment:
```suggestion
            Path(self._ssh_key_file_name).write_text(ssh_key)
```

Nothing wrong with a contextmanager+open, but for small invocations, `pathlib.Path` is nice